sky/shell/gpu/ganesh_canvas.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/shell/gpu/ganesh_canvas.h"
+
+#include "base/logging.h"
+#include "third_party/skia/include/gpu/gl/GrGLInterface.h"
+
+namespace sky {
+namespace shell {
+namespace {
+
+// The limit of the number of GPU resources we hold in the GrContext's
+// GPU cache.
+const int kMaxGaneshResourceCacheCount = 2048;
+
+// The limit of the bytes allocated toward GPU resources in the GrContext's
+// GPU cache.
+const size_t kMaxGaneshResourceCacheBytes = 96 * 1024 * 1024;
+
+}  // namespace
+
+GaneshCanvas::GaneshCanvas() {
+}
+
+GaneshCanvas::~GaneshCanvas() {
+}
+
+void GaneshCanvas::SetGrGLInterface(const GrGLInterface* interface) {
+  sk_surface_.clear();
+  if (interface) {
+    gr_context_ = skia::AdoptRef(GrContext::Create(kOpenGL_GrBackend,
+        reinterpret_cast<GrBackendContext>(interface)));
+    DCHECK(gr_context_);
+    gr_context_->setResourceCacheLimits(kMaxGaneshResourceCacheCount,
+                                        kMaxGaneshResourceCacheBytes);
+  } else if (gr_context_) {
+    gr_context_->abandonContext();
+    gr_context_.clear();
+  }
+}
+
+SkCanvas* GaneshCanvas::GetCanvas(intptr_t fbo, const SkISize& size) {
```

Comment:
if 'fbo' is an OpenGL framebuffer handle then the correct type is GLint aka int32_T